webcompat/webhooks/helpers.py

```diff
@@ -39,6 +39,23 @@ def extract_browser_label(body):
         return None
 
 
+def extract_extra_label(body):
+    """Parse the "extra label" from the body in comment:
+
+    <!-- @extra_label: value -->.
+    Currently this only handles a single label,
+    because that's all that we set in webcompat.com.
+    """
+    match_list = re.search(r'<!--\s@extra_label:\s([\w-]+)\s-->', body)
```

Comment:
So I guess I have forgotten this part here. There was a long comment which is gone. Sniff. 

Let me try to reassess what I was saying. We have two extract using regex for parsing the body. Twice basically. We could change the strategy and code a bit differently.

This is our issue_body. Instead of parsing twice, We could parse once with a more generic function.

```python
>>> issue_body = """
...         <!-- @browser: Firefox 55.0 -->
...         <!-- @ua_header: Mozilla/5.0 (what) Gecko/20100101 Firefox/55.0 -->
...         <!-- @reported_with: web -->
...         <!-- @extra_label: type-media -->
...          
...         **URL**: https://www.example.com/
...         **Browser / Version**: Firefox 55.0
...         <!-- @browser: Chrome 48.0 -->
...         """
>>> match_list = re.findall(r'<!--\s@(\w+):\s(.+)\s-->', issue_body)
>>> match_list
[('browser', 'Firefox 55.0'), ('ua_header', 'Mozilla/5.0 (what) Gecko/20100101 Firefox/55.0'), ('reported_with', 'web'), ('extra_label', 'type-media'), ('browser', 'Chrome 48.0')]
```

This would return a list of tuples that we could use to extract the data for browser name etc. 

but there is better, this is a list of tuples with unique names.

```python
>>> dict(match_list)
{'reported_with': 'web', 'extra_label': 'type-media', 'ua_header': 'Mozilla/5.0 (what) Gecko/20100101 Firefox/55.0', 'browser': 'Chrome 48.0'}
>>> metadata = dict(match_list)
>>> metadata['extra_label']
'type-media'
```

You have a dictionary fully accessible. 
This would require to modify the tests. but could be a nice way of doing.

Reply:
I like it.